packages/dds/tree/src/edit-manager/editManager.ts

```diff
@@ -128,44 +132,65 @@ export class EditManager<TChangeset, TChangeFamily extends ChangeFamily<any, TCh
             return commitToRebase.changeset;
         }
 
+        let nextRevisionNum = -1;
         const changeRebasedToRef = branch.localChanges.reduceRight(
             (newChange, branchCommit) =>
                 this.changeFamily.rebaser.rebase(
                     newChange,
-                    this.changeFamily.rebaser.invert(branchCommit.changeset),
+                    {
+                        revision: brand(nextRevisionNum--),
+                        change: this.changeFamily.rebaser.invert(changeWithMetadataFromCommit(branchCommit)),
+                    },
                 ),
             commitToRebase.changeset,
         );
 
-        return this.rebaseOverCommits(changeRebasedToRef, this.getCommitsAfter(branch.refSeq));
+        const fullyRebasedChange = this.rebaseOverCommits(changeRebasedToRef, this.getCommitsAfter(branch.refSeq));
+
+        const shouldRemoveReference = (revision: RevisionTag): boolean =>
+            isTemporaryRevision(revision) || isBranchLocalRevision(branch, revision);
```

Comment:
We discussed how it may not be OK to remove references to local branch changes if those are the version of the changes that ends up on the trunk.
This function doesn't explicitly put the produced changes on the trunk but `addSequencedChange`) does put the returned change on the trunk, so we may want to avoid removing those references here.